src/nrniv/nrnbbcore_write.cpp

```diff
@@ -144,6 +146,11 @@ char* (*nrnpy_nrncore_arg_p_)(double tstop);
 typedef void (*bbcore_write_t)(double*, int*, int*, int*, double*, Datum*, Datum*, NrnThread*);
 extern bbcore_write_t* nrn_bbcore_write_;
 
+// nrnthreads_type_return needs deferred_type2artdata when
+// there are threads and artificial cells
+typedef std::vector<std::map<int, std::pair<int, double**> > > Deferred_Type2ArtData;
```

Comment:
Just a side note, they fixed `>` handling in C++11, so this would compile:
 `std::vector<std::map<int, std::pair<int, double**>>>`